lib/worker.js

```diff
@@ -0,0 +1,35 @@
+const { isMainThread, parentPort, workerData } = require('worker_threads')
+const track = require('./progressTracker')
+const run = require('./run')
+
+function runTracker (argv, ondone) {
+  const tracker = run(argv)
+
+  tracker.on('done', (result) => {
+    if (ondone) ondone(result)
+    if (argv.json) {
+      console.log(JSON.stringify(result))
+    }
+  })
+
+  tracker.on('error', (err) => {
+    if (err) {
+      throw err
+    }
+  })
+
+  // if not rendering json, or if std isn't a tty, track progress
+  if (!argv.json || !process.stdout.isTTY) track(tracker, argv)
+
+  process.once('SIGINT', () => {
+    tracker.stop()
+  })
+}
+
+if (!isMainThread) {
+  const { opts } = workerData
+
+  runTracker(opts, (result) => {
+    parentPort.postMessage(result)
+  })
```

Comment:
Unfortunately I fear this won't work. I don't think it's possible to combine multiple histograms into one. We need to put all the data in the same histogram, i.e. the workers should send the samples in.